### PR TITLE
Cleanup - eslint fixes - utils actions

### PR DESCRIPTION
**Pull request by lb-**
* part of #8731 - split from #8578
* move `identity` to its own util file (this is a really useful function) & add unit tests for it
* fix the only linting issue in utils/actions.ts - prefer-default-first with a small change
* add a basic set of unit tests for actions.ts

### CODE REVIEW

**Comment by lb-** [comment]
Rebased

**Comment by lb-** [comment]
Rebased

**Comment by lb-** [comment]
Rebased

**Review by thibaudcolas — APPROVED** [review]
Changes to the implementation look good (yay for getting rid of that `func` parameter). There are things I’m surprised about between the tests and types, will try to see if it’s simple enough to fix when merging.

File: client/src/utils/actions.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { createAction } from './actions';
+
+describe('createAction', () => {
+  it('should return a function that creates an action that returns an object', () => {
+    expect(createAction()()).toEqual({});
+  });
+
+  it('should accept an action creator and return a function that creates an action', () => {
+    const actionCreator = jest.fn(() => ({ data: true }));
```
[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/actions.test.js:9]: Happy to remove these tests - my thinking here and above is it's good to test potential rough edge cases to ensure the code does not explode.

However, that may be just the way I work with my code at work and might not be worth the overhead in these unit tests.

Let me know if I should simplify these unit tests to just cover the happy path only.

[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/actions.test.js:9]: ok I see what the confusion is here, we actually do check that it got called with params (a few lines below) but I will update the test nonetheless.

File: client/src/utils/actions.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { createAction } from './actions';
+
+describe('createAction', () => {
+  it('should return a function that creates an action that returns an object', () => {
+    expect(createAction()()).toEqual({});
```
[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/actions.test.js:5]: ok I will remove this test for now, this is the current behaviour and I think that changing it should not be in scope of this PR.

I do not know enough about this code and I am pretty sure that Redux will not be happy if we accidentally let non-objects leak into the actions.

From my knowledge an empty object is ok though, it will just do nothing.

However, I will remove this unit test and merge this PR in.

[the remark this answers is not in the document]
Comment (thibaudcolas) [reply client/src/utils/actions.test.js]: 👍 all I meant was switching to:

```
expect(createAction('SOME_ACTION_NAME')()).toEqual({ type: 'SOME_ACTION_NAME' });
```

so we technically follow the signature of the function. So it’s only the tests that need changing. Shouldn’t we be writing those tests in TypeScript as well? This way those issues would become apparent from the get-go.

[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/actions.test.js]: Ok all good - I'll leave this as is though - we can add more tests later if needed. 

My understanding is that we write tests in JavaScript - not TypeScript - I assumed that by convention and also added to the work in progress UI development docs. 

But I'm not sure whether I actually heard that anywhere. 

I'm personally a fan of JS unit tests as it allows you to write things that 'should not happen' in TS - remembering we want to ensure some code is more robust than just our types will allow. 

I have also historically found TS tests add a lot of overhead to write with little benefits in other projects. 

I don't think we have any unit tests in TS right now - but if that's something we want to do we should maybe add to the UI team agenda.

[the remark this answers is not in the document]
Comment (lb-) [reply client/src/utils/actions.test.js]: I've also told other contributors not to write unit tests in TypeScript (either in some if the Outreachy issues or on PRs). But happy to align with whatever we want to do in the project - would be good to document the preferred approach once we all agree.
